site/src/pages/WorkspacePage/ResourcesSidebar.tsx

```diff
@@ -12,13 +12,13 @@ import { getResourceIconPath } from "utils/workspace";
 type ResourcesSidebarProps = {
   failed: boolean;
   resources: WorkspaceResource[];
-  onChange: (resourceId: string) => void;
-  selected: string;
+  onChange: (resource: WorkspaceResource) => void;
+  isSelected: (resource: WorkspaceResource) => boolean;
```

Comment:
I think the `isSelected` change is an improvement, but I'm wondering if the implementation is getting a little too tied to how `useResourcesNav` is set up.

What do you think of turning `isSelected` into just a boolean, and making the parent component call the function to get the right prop to pass in?

Reply:
We need to know what is the selected resource and since this is not just an attribute but a "calculation", I think this interface makes things easier to extend.